src/coreclr/src/gc/gcpriv.h

```diff
@@ -4695,7 +4702,10 @@ class heap_segment
     uint8_t*        background_allocated;
 #ifdef MULTIPLE_HEAPS
     gc_heap*        heap;
+    uint8_t*        saved_committed;
+    size_t          saved_desired_allocation;
```

Comment:
I would make these _DEBUG only fields and assert on them - it's nice to not have to increase the size of `heap_segment` in release and these things really should be guarantees so asserts are more appropriate.

Reply:
Done.